### ISSUE TITLE
SlicerAutomatedDentalTools Extension Not Available in Slicer 5.6.1 Stable and Nightly Builds

### ISSUE THREAD

**Issue by luciacev** [DCBIA-OrthoLab/SlicerAutomatedDentalTools#59]
@allemangD @bpaniagua @jcfr @lassoan
### Issue Description
**Extension Name:** SlicerAutomatedDentalTools
**Affected Slicer Version:** 5.6.1 Stable and Nightly Builds
**Operating Systems:** Linux, Windows, Mac

#### Problem
The SlicerAutomatedDentalTools extension, which includes modules for Automated CBCT and Intra-oral Scans (such as AOS, AReg, AMASSS, AutoMatrix, and AutoCrop3D), is not available for download in the Slicer 5.6.1 stable version nor in the nightly builds since last week. This extension is critical for our workflows, and its absence in the latest versions is impacting our operations.

#### Expected Behavior
In previous versions (e.g., Slicer 5.6), this extension was available and functional. We expect the extension to be accessible and operational in both the Slicer 5.6.1 stable release and the nightly versions.

#### Steps to Reproduce
1. Open 3D Slicer version 5.6.1 (stable or nightly build).
2. Navigate to the Extension Manager.
3. Search for "SlicerAutomatedDentalTools".
4. Observe that the extension does not appear in the search results.

#### Additional Information
- The extension works as expected in Slicer version 5.6.
- @bianchijonas1 first reported this issue to me on Friday and @Jeanneclre  @GaelleLeroux do not know how to address this issue

#### Request
We request assistance in resolving this issue to ensure that the SlicerAutomatedDentalTools extension is available and functional in Slicer 5.6.1 stable and nightly builds. Any guidance on troubleshooting or timelines for a resolution would be greatly appreciated.

---

Thank you for your attention to this matter and for your support in maintaining the functionality of essential extensions within the Slicer community.

**Comment by bpaniagua** [comment]
@luciacev thank you for reporting this!
@allemangD could you please look into this?

**Comment by jcfr** [comment]
The extension is failing to build on all plaforms for both Slicer `Stable` (5.6.1) and `Preview` (~5.7.0):

| | Build report of `2023-12-20` |
|--|--|
| `SlicerPreview` | https://slicer.cdash.org/index.php?project=SlicerPreview&date=2023-12-20&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=AutomatedDentalTools |
| `SlicerStable` | https://slicer.cdash.org/index.php?project=SlicerStable&date=2023-12-20&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=AutomatedDentalTools |


For example, on Windows, the reported error[^1] is the following:

```
CMake Error at D:/D/P/S-0/CMake/SlicerMacroBuildScriptedModule.cmake:78 (message):
  slicerMacroBuildScriptedModule(SCRIPTS) given nonexistent file or directory
  'utils_windows/__init__.py'
Call Stack (most recent call first):
  ALI_IOS/CMakeLists.txt:17 (slicerMacroBuildScriptedModule)
```

### Notes

To avoid introducing regression in extension being distributed, having a locally build may be helpful to locally confirm the extension can effectively be packaged.

To help clarify this and explain where and how to identify these type of issues, this pull request clarifying the Slicer documentation should help:
* https://github.com/Slicer/Slicer/pull/7501

Would appreciate if you team could review.

### Path forward

Given the error posed above, it should be easy to fix the build-system to re-enable package distribution.

For convenience, I will look into submitting a fix for addressing it.

[^1]: https://slicer.cdash.org/build/3244400/configure

**Comment by jcfr** [comment]
Related to the `ALI_IOS` module:
* https://github.com/DCBIA-OrthoLab/SlicerAutomatedDentalTools/issues/48

**Comment by jcfr** [comment]
Related to `ALI & AMASS module`:
* https://github.com/DCBIA-OrthoLab/SlicerAutomatedDentalTools/issues/4

**Comment by Jeanneclre** [comment]
Thank you for the feedback.
If I've understood correctly, for code developed in python we're not obliged to have a locally build but it's recommended, right?

**Comment by luciacev** [comment]
@allemangD  @jcfr @bpaniagua  @Jeanneclre  @GaelleLeroux

We'd like to provide an update on our efforts to address the issues with the SlicerAutomatedDentalTools extension:

Actions Taken and Findings:
1. Review of Build Errors: We've carefully reviewed the build reports on CDash for both the stable and preview versions. This allowed us to identify specific errors causing the build process to fail across various platforms.
2. Addressing the Specific Error: We've pinpointed the error related to the missing utils_windows/__init__.py file. We have verified its presence and proper referencing within our build configuration files.
3. Local Build and Debugging: @Jeanneclre and @GaelleLeroux are in the process of initiating local builds and debugging using Slicer's debug build. We believe this will be key in providing immediate feedback and insights into the underlying issues. There's an indication that the ALI_IOS module may be a contributing factor, and @GaelleLeroux is currently addressing this matter.
4. Reviewing Documentation and Pull Request: We have gone through the recommended updates in the Slicer documentation, as outlined in pull request Slicer/Slicer#7501. We are actively applying the suggested changes and strategies.
5. Collaboration and Assistance: We intend to reach out to @allemangD for further assistance to support our troubleshooting efforts.
6. Next Steps: We will persist with our independent troubleshooting and eagerly anticipate any patches or fixes from @allemangD  @jcfr , which may streamline our resolution of these issues.
